### PR TITLE
i3c_controller: Add IBI corner case test

### PR DESCRIPTION
**Pull request by gastmaier**
Test when the controller is listening for IBIs but the feature itself is disabled.
Regenerate adi_regmap_i3c_controller_pkg.sv using the perl script.

### CODE REVIEW

File: i3c_controller/tests/test_program.sv [hunk 1]
```diff
@@ -785,6 +785,23 @@ task ibi_i3c_test();
...
+  axi_write (`I3C_CONTROLLER_BA, `I3C_REGMAP_IBI_CONFIG, 2'b10);
+
+  #12000ns
+  `INFO(("IBI I3C Test #6"));
+  auto_ack <= 1'b0;
+  write_ibi_da(DEVICE_DA1+1);
+
+  #10000ns
+  if (`DUT_I3C_REGMAP.ibi_fifo_valid)
+    `ERROR(("IBI should not have thrown IBI"));
```
Comment (IstvanZsSzekely) [inline i3c_controller/tests/test_program.sv:803]: Is the Valid going to stay on if IBI throws IBI? Typo?

Comment (gastmaier) [reply i3c_controller/tests/test_program.sv:803]: The test comment is clearer than the error description, but in summary, the error is interpreted as:
"The In-band interrupt that occurred on the bus shouldn't have generated an entry in the IBI FIFO since it was not acknowledged by the controller"
Only acknowledged IBI requests create an entry in the IBI FIFO.

> Is the Valid going to stay on if IBI throws IBI?
If the IBI on the bus is acknowledged, yes, until the FIFO is read (tests 1,2,5).

Comment (IstvanZsSzekely) [reply i3c_controller/tests/test_program.sv:803]: Okay, that makes sense, and I got both answers for my questions.

**Review by IstvanZsSzekely — APPROVED** [review]
